api/filter/http/gzip.proto

```diff
@@ -46,41 +39,33 @@ message Gzip {
     RLE = 3;
   }
 
-  // Allows selecting zlib's compression strategy. Strategy is directly
-  // related to the characteristics of the content which is being compressed. Most of the time
-  // "DEFAULT" will be the best choice, however there are situations which changing the strategy
-  // might produce better results. For example, Run-length encoding (RLE) is normally used when the
-  // content is known for having sequences which same data occurs many consecutive times. For more
-  // information about each strategy, please refer to Zlib manual. This field will be set to
-  // "DEFAULT" if not specified.
+  // A value used for selecting Zlib's compression strategy which is directly related to the
+  // characteristics of the content. Most of the time "DEFAULT" will be the best choice, though
+  // there are situations which changing this parameter might produce better results. For example,
+  // Run-length encoding (RLE) is typically used when the content is known for having sequences
+  // which same data occurs many consecutive times. For more information about each strategy, please
+  // refer to Zlib manual.
   CompressionStrategy compression_strategy = 4 [(validate.rules).enum.defined_only = true];
 
-  // Array of strings that allows specifying which "cache-control" header
-  // values yield compression. Normally, if "cache-control" is present in the response headers,
-  // compression should only occur if directives indicate that the content should not be cached;
-  // e.g. no-cache or no-store.
-  repeated string cache_control = 5
-      [(validate.rules).repeated = {min_items: 0, max_items: 10, unique: true}];
-
-  // Array of strings that allows specifying which mime-types yield compression; e.g.
-  // application/json, text/html, etc. When this field is not specified, compression will be applied
-  // to any "content-type".
-  repeated string content_type = 6 [(validate.rules).repeated = {min_items: 0, max_items: 30}];
+  // Set of strings that allows specifying which mime-types yield compression; e.g.,
+  // application/JSON, text/HTML, etc. When this field is not defined, compression will be applied
+  // to the following mime-types: "application/javascript", "application/json",
+  // "application/xhtml+xml", "image/svg+xml", "text/css", "text/html", "text/plain", "text/xml".
+  repeated string content_type = 6 [(validate.rules).repeated = {min_items: 0, max_items: 50}];
 
-  // Allows disabling compression if response contains "etag" (entity tag)
-  // header. Default is false.
-  google.protobuf.BoolValue disable_on_etag = 7;
+  // If true, disables compression when the response contains Etag header. When it is false, the
+  // filter will preserve weak Etags and remove the ones that require strong validation. Note that
+  // if any of these options fit the specific scenario, an alternative solution is to disabled Etag
+  // at the origin and use Last-Modified header instead.
```

Comment:
Nit: prefer lower case header names (for H2 consistency) and the RST style `*last-modified*`.